### PR TITLE
Removing v1alpha2 directory from docs

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
The v1alpha2 directory in our docs existed to prevent any old/outdated links from breaking, but I think we've outlived it's usefulness. I'd expect that anything linking to an old v1alpha2 URL is likely several years out of date. This removes that older structure to simplify managing our docs. Thanks to @candita for catching this one!

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: site-src/v1alpha2/guides
Comment (candita) [inline site-src/v1alpha2/guides:1]: is there still a TLS guide to moved?  It used to be in v1alpha2 but I don't see it moved in this PR.

File: geps/gep-1897/index.md [hunk 1]
```diff
@@ -600,7 +600,7 @@ the implementation would be required to fully implement the policy or mark the b
 
 [Policy Attachment](https://gateway-api.sigs.k8s.io/reference/policy-attachment/#direct-policy-attachment)
 
-[Gateway API TLS](https://gateway-api.sigs.k8s.io/v1alpha2/guides/tls/)
+[Gateway API TLS](https://gateway-api.sigs.k8s.io/guides/tls/)
```
Comment (candita) [inline geps/gep-1897/index.md:603]: Did this need to move?  Should it be added to this PR in another folder?

File: site-src/v1alpha2/guides
Comment (robscott) [reply site-src/v1alpha2/guides:1]: Nope, these were all just symlinks so everything in the directory would have been a copy with the main guides directory.

Comment (candita) [reply site-src/v1alpha2/guides:1]: Never mind, I see that these were just redirects.

**Review by shaneutt — APPROVED** [review]
/approve

**Comment by robscott** [comment]
I think Prow is confused, I just rebased this and GitHub UI is not asking for rebase. Maybe this will wake it up.

/honk
/hold cancel

**Pull request merged**